.github/workflows/ci.yml

```
@@ -0,0 +1,118 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  merge_group:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  change-detection:
+    name: 🔍 Change
+    uses: cda-tum/mqt-workflows/.github/workflows/reusable-change-detection.yml@v1.7
+
+  python-tests:
+    name: 🐍 Test
+    needs: change-detection
+    if: fromJSON(needs.change-detection.outputs.run-python-tests)
+    uses: cda-tum/mqt-workflows/.github/workflows/reusable-python-ci.yml@v1.7
+    permissions:
+      contents: read
+      id-token: write
+    with:
+      # Runs to enable
+      enable-ubuntu2404: true
+      enable-ubuntu2404-arm: true
+      enable-macos13: true
+      enable-macos14: true
+      enable-windows2022: true
+      # Runs to disable
+      enable-ubuntu2204: false
+      enable-ubuntu2204-arm: false
+      enable-macos15: false
+      enable-windows2025: false
+
+  # run extensive Python tests whenever this is on a PR and the PR has the `extensive-python-ci` label
+  python-tests-extensive:
+    name: 🐍 Test (Extensive)
+    needs: change-detection
+    if: fromJSON(needs.change-detection.outputs.run-python-tests) && github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'extensive-python-ci')
+    uses: cda-tum/mqt-workflows/.github/workflows/reusable-python-ci.yml@v1.7
+    permissions:
+      contents: read
+      id-token: write
+    with:
+      # Runs to enable
+      enable-ubuntu2204: true
+      enable-ubuntu2204-arm: true
+      enable-macos15: true
+      enable-windows2025: true
+      # Runs to disable (these are already enabled in the non-extensive run)
+      enable-ubuntu2404: false
+      enable-ubuntu2404-arm: false
+      enable-macos13: false
+      enable-macos14: false
+      enable-windows2022: false
+
+  python-linter:
+    name: 🐍 Lint
+    needs: change-detection
+    if: fromJSON(needs.change-detection.outputs.run-python-tests)
+    uses: cda-tum/mqt-workflows/.github/workflows/reusable-python-linter.yml@v1.7
+
+  code-ql:
+    name: 📝 CodeQL
+    needs: change-detection
+    if: fromJSON(needs.change-detection.outputs.run-code-ql)
+    uses: cda-tum/mqt-workflows/.github/workflows/reusable-code-ql.yml@v1.7
```

```suggestion
    uses: cda-tum/mqt-workflows/.github/workflows/reusable-code-ql-python.yml@v1.7
```
Should resolve one of the CI failures. That was an oversight on my part.